src/pybind/mgr/dashboard/controllers/cluster.py

```diff
@@ -0,0 +1,23 @@
+# -*- coding: utf-8 -*-
+
+from .. import mgr
+from ..services import cluster_service
+from . import ApiController, ControllerDoc, Endpoint, EndpointDoc, RESTController
+
+
+@ApiController('/cluster', secure=False)
+@ControllerDoc("Get Cluster Details", "Cluster")
+class Cluster(RESTController):
+    @Endpoint('GET')
+    @EndpointDoc("Get the cluster status")
+    def status(self):
+        return cluster_service.fetch_status()
+
+    @Endpoint('PUT')
+    @EndpointDoc("Update the cluster status",
+                 parameters={'cluster_status': (str, 'Cluster Status')})
+    def update_status(self, cluster_status: str):
+        for status in cluster_service.ClusterStatus:
+            if status.name == cluster_status:
+                mgr.set_store('cluster-status', cluster_status)
+                return
```

Comment:
What about using a single resource (`Cluster`) with a data model (`ClusterModel`)? We could even think about serializing the class directly into the mon KV store (with JSON or pickle):

```py
# This needs Python 3.8 but CentOS supports it

from dataclasses import dataclass
from enum import Enum


@dataclass
class ClusterModel:
	class Status(Enum):
	    INSTALLED = 0
	    POSTINSTALLED = 1

    status: Status


@ApiController('/cluster', secure=False)
@ControllerDoc("Get Cluster Details", "Cluster")
class Cluster(RESTController):
    @EndpointDoc("Get the cluster status")
    def list(self):
        return ClusterModel(status=mgr.get_store('cluster/status', default=...))

    @EndpointDoc("Update the cluster status",
                 parameters={'status': (str, 'Cluster Status')})
    def set(self, cluster: Cluster):
        mgr.set_store('cluster/status', cluster.status)

```

Reply:
In progress...